src/stretchy.js

```diff
@@ -52,130 +52,512 @@ const mathMLnode = function(label) {
+
+// Nested SVGs
+//    Many of the KaTeX SVG images contain a nested SVG. This is done to
+//    achieve a stretchy image while avoiding distortion of arrowheads or
+//    brace corners.
+
+//    The inner SVG typically contains a very long (400 em) arrow.
+
+//    The outer SVG acts like a window that exposes only part of the inner SVG.
+//    The outer SVG will grow or shrink to match the dimensions set by CSS.
+
+//    The inner SVG always has a longer, thinner aspect ratio than the outer
+//    SVG. After the inner SVG fills 100% of the height of the outer SVG,
+//    there is a long arrow shaft left over. That left-over shaft is not shown.
+//    Instead, it is sliced off because the inner SVG is set to
+//    "preserveAspectRatio='... slice'".
+
+//    Thus, the reader sees an arrow that matches the subject matter width
+//    without distortion.
+
+//    Some functions, such as \cancel, need to vary their aspect ratio. These
+//    functions do not get the nested SVG treatment.
+
+// Second Brush Stroke
+//    Low resolution monitors struggle to display images in fine detail.
+//    So browsers apply anti-aliasing. A long straight arrow shaft therefore
+//    will sometimes appear as if it has a blurred edge.
+
+//    To mitigate this, these SVG files contain a second "brush-stroke" on the
+//    arrow shafts. That is, a second long thin rectangular SVG path has been
+//    written directly on top of each arrow shaft. This reinforcement causes
+//    some of the screen pixels to display as black instead of the anti-aliased
+//    gray pixel that a  single path would generate. So we get arrow shafts
+//    whose edges appear to be sharper.
+
+const innerSVG = {
+    // Since bcancel's SVG is inline and it omits the viewBox attribute,
+    // the browswer will render stroke-width='1.5px' in the screen context.
+    bcancel: `<line x1='0' y1='0' x2='100%' y2='100%' stroke-width='1.5px'
+stroke='currentcolor'/>`,
+
+    cancel: `<line x1='0' y1='100%' x2='100%' y2='0' stroke-width='1.5px'
+stroke='currentcolor'/>`,
+
+    // The doubleleftarrow geometry is from glyph U+21D0 in the font KaTeX Main
+    doubleleftarrow: `><svg viewBox='0 0 400000 549'
+preserveAspectRatio='xMinYMin slice'><path fill='currentcolor' d='M262 157
+l10-10c34-36 62.7-77 86-123 3.3-8 5-13.3 5-16 0-5.3-6.7-8-20-8-7.3
+ 0-12.2.5-14.5 1.5-2.3 1-4.8 4.5-7.5 10.5-49.3 97.3-121.7 169.3-217 216-28
+ 14-57.3 25-88 33-6.7 2-11 3.8-13 5.5-2 1.7-3 4.2-3 7.5s1 5.8 3 7.5
+c2 1.7 6.3 3.5 13 5.5 68 17.3 128.2 47.8 180.5 91.5 52.3 43.7 93.8 96.2 124.5
+ 157.5 9.3 8 15.3 12.3 18 13h6c12-.7 18-4 18-10 0-2-1.7-7-5-15-23.3-46-52-87
+-86-123l-10-10h399738v-40H218c328 0 0 0 0 0l-10-8c-26.7-20-65.7-43-117-69 2.7
+-2 6-3.7 10-5 36.7-16 72.3-37.3 107-64l10-8h399782v-40z
+m8 0v40h399730v-40zm0 194v40h399730v-40z'/></svg>`,
+
+    // doubleleftrightarrow is from glyph U+21D4 in font KaTeX Main
+    doubleleftrightarrow: `><svg width='50.1%' viewBox='0 0 400000 549'
```

Comment:
If we try for an exact 50%-50% match, then floating point rounding error can sometimes create a 1 screen-pixel vertical white line in the middle of the arrow shaft. Setting one side to 50.1% is enough overlap to avoid this. Except one element, I think the braces, where Firefox needed something larger.